examples/sha256/table16/compression/compression_util.rs

```diff
@@ -38,101 +38,155 @@ pub const SUBREGION_MAIN_WORD: usize =
     DECOMPOSE_ABCD + SIGMA_0_ROWS + DECOMPOSE_EFGH + SIGMA_1_ROWS + CH_ROWS + MAJ_ROWS;
 pub const SUBREGION_MAIN_ROWS: usize = SUBREGION_MAIN_LEN * SUBREGION_MAIN_WORD;
 
+/// Round index.
+#[derive(Debug, Copy, Clone)]
+pub enum RoundIdx {
+    Init,
+    Main(usize),
+}
+
+impl RoundIdx {
+    pub(crate) fn as_usize(&self) -> usize {
+        match self {
+            Self::Main(idx) => *idx,
+            _ => panic!(),
+        }
+    }
+}
```

Comment:
Given how many methods panic on `RoundIdx::Init`, I would define a separate `MainRound` struct for them, and leave the match on `RoundIdx` to the caller. Non-blocking, I'm interested in doing this myself in a separate PR.